HIP-Basic/runtime_compilation/README.md

```diff
@@ -0,0 +1,91 @@
+# HIP-Basic Runtime Compilation Example
+
+## Description
+Runtime compilation allows compiling code to machine code while a program is being executed rather than before, as happens with static compilation.
+HIP supports runtime compilation (hipRTC), allowing to compile certain fragments of code in runtime and thus permitting runtime optimizations in them.
+Therefore, the usage of hipRTC provides the possibility of obtaining optimizations and performance improvements compared with other APIs with regular offline static compilation.
+
+This example showcases how to make use of hipRTC to compile in runtime a kernel and launch it on a device. This kernel is a simple SAXPY, i.e. a single-precision operation $y_i=ax_i+y_i$.
+
+### Application flow 
+1. A number of variables are declared and defined to configure the program which will be compiled in runtime.
+2. The program is created using the above variables as parameters, along with the SAXPY kernel in string form.
+3. The properties of the first device (GPU) available are consulted to set the device architecture as (the only) compile option.
+4. The program is compiled using the previously mentioned compile options.
```

Comment:
Let's have a flow diagram

Reply:
Added a flow diagram in HIP-Basic/runtime_compilation/hiprtc.svg. You can see a preview [here](https://github.com/amd/rocm-examples/tree/53bad130d49b27421e32478b1194a8723c0d2edc/HIP-Basic/runtime_compilation).

Reply:
nice